lib/widgets/compose_box.dart

```diff
@@ -1128,8 +1110,37 @@ class ComposeBox extends StatelessWidget {
     }
   }
 
+  Widget? _errorBanner(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final selfUser = store.users[store.selfUserId]!;
+    switch (narrow) {
+      case ChannelNarrow narrow:
+        final channel = store.streams[narrow.streamId]!;
+        return channel.hasPostingPermission(selfUser, byDate: DateTime.now(), realmWaitingPeriodThreshold: store.realmWaitingPeriodThreshold)
+          ? null : _ErrorBanner(label: ZulipLocalizations.of(context).errorBannerCannotPostInChannelLabel);
+      case TopicNarrow narrow:
+        final channel = store.streams[narrow.streamId]!;
+        return channel.hasPostingPermission(selfUser, byDate: DateTime.now(), realmWaitingPeriodThreshold: store.realmWaitingPeriodThreshold)
+          ? null : _ErrorBanner(label: ZulipLocalizations.of(context).errorBannerCannotPostInChannelLabel);
+      case DmNarrow(:final otherRecipientIds):
+        final hasDeactivatedUser = otherRecipientIds.any((id) =>
+          !(store.users[id]?.isActive ?? true));
+        return hasDeactivatedUser ? _ErrorBanner(label: ZulipLocalizations.of(context)
+          .errorBannerDeactivatedDmLabel) : null;
```

Comment:
nit: to make the lines shorter:
```suggestion
        if (hasDeactivatedUser) {
          return _ErrorBanner(
            label: localizations.errorBannerDeactivatedDmLabel);
        }
```